src/Integrations/Integrations/CakePHP/V3/CakePHPIntegrationLoader.php

```diff
@@ -0,0 +1,156 @@
+            $integration->rootSpan->service = $integration->appName;
+            if ('cli' === PHP_SAPI) {
+                $integration->rootSpan->name = 'cakephp.console';
+                $integration->rootSpan->resource =
+                    !empty($_SERVER['argv'][1]) ? 'cake_console ' . $_SERVER['argv'][1] : 'cake_console';
+            } else {
+                $integration->rootSpan->name = 'cakephp.request';
+                $integration->rootSpan->meta[Tag::SPAN_KIND] = 'server';
+            }
+            $integration->rootSpan->meta[Tag::COMPONENT] = CakePHPIntegration::NAME;
+        };
+
+        \DDTrace\hook_method('App\Application', '__construct', $setRootSpanInfoFn);
+        \DDTrace\hook_method('Cake\Http\Server', '__construct', $setRootSpanInfoFn);
+
+        \DDTrace\trace_method(
+            'Cake\Controller\Controller',
+            'invokeAction',
+            function (SpanData $span) use ($integration) {
+                $span->name = $span->resource = 'Controller.invokeAction';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $integration->appName;
+                $span->meta[Tag::COMPONENT] = CakePHPIntegration::NAME;
+
+                /** @var \Cake\Controller\Controller $this */
+                $request = $this->request;
+                if (!$request instanceof ServerRequest) {
+                    return;
+                }
+
+                if (dd_trace_env_config("DD_HTTP_SERVER_ROUTE_BASED_NAMING")) {
+                    $integration->rootSpan->resource =
+                        $_SERVER['REQUEST_METHOD'] . ' ' . $this->name . 'Controller@' . $request->getParam('action');
+                }
+
+                if (!array_key_exists(Tag::HTTP_URL, $integration->rootSpan->meta)) {
+                    $integration->rootSpan->meta[Tag::HTTP_URL] = Router::url($request->getAttribute('here'), true)
+                        . Normalizer::sanitizedQueryString();
+                }
+                $integration->rootSpan->meta['cakephp.route.controller'] = $request->getParam('controller');
+                $integration->rootSpan->meta['cakephp.route.action'] = $request->getParam('action');
+                $plugin = $request->getParam('plugin');
+                if ($plugin) {
+                    $integration->rootSpan->meta['cakephp.plugin'] = $plugin;
+                }
+            }
+        );
+
+        \DDTrace\trace_method('Cake\Error\Middleware\ErrorHandlerMiddleware', 'handleException', [
+            'instrument_when_limited' => 1,
+            'posthook' => function (SpanData $span, array $args) use ($integration) {
+                $integration->setError($integration->rootSpan, $args[0]);
+                $span->meta[Tag::COMPONENT] = CakePHPIntegration::NAME;
+                return false;
+            },
+        ]);
+
+        \DDTrace\trace_method('Cake\Http\Response', 'getStatusCode', [
```

Comment:
I suppose you copied from V2, but shouldn't we just use proper hook_method for these? I mean `return false` just unconditionally drops the span...

I think we can improve that one in all instances here.

Reply:
Absolutely :) Copy-pasted as you said and didn't pay attention to this; I'll change these